features/users/authenticate_users.feature

```diff
@@ -25,48 +25,119 @@ Feature: User Authentication
-    And I fill in "User name" with "sam"
-    And I fill in "sam"'s temporary password
-    And I press "Log In"
+      And I fill in "User name" with "sam"
+      And I fill in "sam"'s temporary password
+      And I press "Log In"
     Then I should see "Hi, sam"
-    And I should see "Change My Password"
+      And I should see "You used a temporary password to log in."
+      And I should see "Change My Password"
 
     # and I should be able to change the password
     When I fill in "New password" with "newpass"
-    And I fill in "Confirm new password" with "newpass"
-    And I press "Change Password"
+      And I fill in "Confirm new password" with "newpass"
+      And I press "Change Password"
     Then I should see "Your password has been changed"
 
     # old password should no longer work
     When I am logged out
     When I am on the homepage
-    And I fill in "User name" with "sam"
-    And I fill in "Password" with "secret"
-    And I press "Log In"
+      And I fill in "User name" with "sam"
+      And I fill in "Password" with "secret"
+      And I press "Log In"
     Then I should not see "Hi, sam"
 
     # generated password should no longer work
     When I am logged out
     When I am on the homepage
-    And I fill in "User name" with "sam"
-    And I fill in "sam"'s temporary password
-    And I press "Log In"
+      And I fill in "User name" with "sam"
+      And I fill in "sam"'s temporary password
+      And I press "Log In"
     Then I should not see "Hi, sam"
 
     # new password should work
     When I am logged out
     When I am on the homepage
-    And I fill in "User name" with "sam"
-    And I fill in "Password" with "newpass"
-    And I press "Log In"
+      And I fill in "User name" with "sam"
+      And I fill in "Password" with "newpass"
+      And I press "Log In"
+    Then I should see "Hi, sam"
+
+  Scenario: Forgot password, logging in with email address
+    Given I have no users
+      And the following activated user exists
+        | login | email           | password |
+        | sam   | sam@example.com | password |
+      And all emails have been delivered
+    When I am on the login page
+      And I follow "Reset password"
+      And I fill in "reset_password_for" with "sam@example.com"
+      And I press "Reset Password"
+    Then 1 email should be delivered
+    When I am logged out
+      And I fill in "User name" with "sam@example.com"
+      And I fill in "sam"'s temporary password
+      And I press "Log In"
     Then I should see "Hi, sam"
+      And I should see "You used a temporary password to log in."
+      And I should see "Change My Password"
+
+  Scenario: With expired password token
+    Given I have no users
+      And the following activated user exists
+        | login | password |
+        | sam   | password |
+      And all emails have been delivered
+    When I am on the login page
+      And I follow "Reset password"
+      And I fill in "reset_password_for" with "sam"
```

Comment:
If we could use the label text ("Email address or user name") instead, that would be great. (If it doesn't work, we might have an HTML booboo.)